xml/System.Runtime.CompilerServices/Unsafe.xml

```diff
@@ -407,12 +488,33 @@
         <Parameter Name="source" Type="TFrom" RefType="ref" Index="0" FrameworkAlternate="dotnet-plat-ext-2.1;dotnet-plat-ext-2.2;net-5.0;net-6.0;netcore-1.1;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <typeparam name="TFrom">The type of reference to reinterpret.</typeparam>
-        <typeparam name="TTo">The desired type of the reference.</typeparam>
-        <param name="source">The reference to reinterpret.</param>
-        <summary>Reinterprets the given reference as a reference to a value of type <typeparamref name="TTo" />.</summary>
-        <returns>A reference to a value of type <typeparamref name="TTo" />.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="TFrom">The type of managed pointer to reinterpret.</typeparam>
+        <typeparam name="TTo">The desired type of the managed pointer.</typeparam>
+        <param name="source">The managed pointer to reinterpret.</param>
+        <summary>Reinterprets the given managed pointer as a new managed pointer to a value of type <typeparamref name="TTo" />.</summary>
+        <returns>A managed pointer to a value of type <typeparamref name="TTo" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This API is conceptually similar to C++'s `reinterpret_cast<>`. It is the caller's responsibility to ensure that the cast is legal. No runtime check will be performed.
+
+Only the managed pointer is reinterpreted. The referenced value itself will remain unchanged. Consider the example below.
+
+```cs
+int[] intArray = new int[] { 0x12345678 }; // a 1-element array
```

Comment:
```suggestion
int[] intArray = new int[] { 0x_1234_5678 }; // a 1-element array
```
For the example below it's easier to follow then.

Reply:
Neat suggestion!

Reply:
nit: I much prefer `0x1234_5678`, the `x_` makes it much harder to read for me.